src/components/Alphabet/Letter/Letter.js

```diff
@@ -1,34 +1,62 @@
 import React from 'react';
 import styles from './Letter.css';
+import classnames from 'classnames';
 
-const checkLetter = (letter, id, onLetterClick) => {
-    const url = `/phrases/check`;
-    return fetch(url, {
-        headers: {
-            'Content-Type': 'application/json',
-            'Accept': 'application/json',
-        },
-        method: 'post',
-        body: JSON.stringify({
-            id,
-            letter,
+class Letter extends React.PureComponent {
+    state = {
+        isLetterCorrect: null,
+        usedLetter: false,
+    };
+
+    checkLetter = (letter, id) => {
+        const url = `/phrases/check`;
+        return fetch(url, {
+            headers: {
+                'Content-Type': 'application/json',
+                'Accept': 'application/json',
+            },
+            method: 'post',
+            body: JSON.stringify({
+                id,
+                letter,
+            })
         })
-    })
-    .then(response => response.json())
-    .catch(error => {
-        console.error(error);
-    })
-    .then(data => {
-        onLetterClick(data);
-    });
-}
+        .then(response => response.json())
+        .catch(error => {
+            console.error(error);
+        })
+        .then(data => {
+            this.setState({
+                isLetterCorrect: data.failsCounter === this.props.failsCounter,
+            })
+            this.props.onLetterClick(data);
+        });
+    }
 
-const Letter = React.memo(({letter, gameID, onLetterClick}) => {
-    return(
-        <div className={styles.letter} onClick={checkLetter.bind(null, letter, gameID, onLetterClick)}>
-            {letter}
-        </div>
-    )
-});
+    onClickMethod = () => {
```

Comment:
I don't know from name what this function does (bad naming)

Reply:
checkLetterOnClick is better?

Reply:
maybe onLetterClick?

Reply:
At the end this function was deleted.